internal/pkg/cli/task_delete.go

```diff
@@ -0,0 +1,380 @@
+	}
+
+	return &deleteTaskOpts{
+		deleteTaskVars: vars,
+
+		store: store,
+		// spinner: termprogress.NewSpinner(log.DiagnosticWriter),
+		prompt: prompter,
+		sess:   provider,
+		sel:    selector.NewWorkspaceSelect(prompter, store, ws),
+		newTaskSel: func(session *awssession.Session) cfTaskSelector {
+			cfn := cloudformation.New(session)
+			return selector.NewCFTaskSelect(prompter, store, cfn)
+		},
+		// newTaskLister: func(session *awssession.Session) tasksLister {
+		// 	return ecs.New(session)
+		// },
+		// newTaskStopper: func(session *awssession.Session) tasksStopper {
+		// 	return ecs.New(session)
+		// },
+		// newTaskDeleter: func(session *awssession.Session) taskDeployer {
+		// 	return cloudformation.New(session)
+		// },
+		// newImageRemover: func(session *awssession.Session) imageRemover {
+		// 	return ecr.New(session)
+		// },
+	}, nil
+}
+
+// Validate checks that flag inputs are valid.
+func (o *deleteTaskOpts) Validate() error {
+
+	if o.name != "" {
+		if err := basicNameValidation(o.name); err != nil {
+			return err
+		}
+	}
+
+	// If default flag specified,
+	if err := o.validateFlagsWithDefaultCluster(); err != nil {
+		return err
+	}
+
+	if err := o.validateFlagsWithEnv(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) validateFlagsWithEnv() error {
+	if o.app != "" {
+		err := o.validateAppName()
+		if err != nil {
+			return err
+		}
+	}
+
+	if o.env != "" {
+		err := o.validateEnvName()
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) validateFlagsWithDefaultCluster() error {
+	if !o.defaultCluster {
+		return nil
+	}
+
+	// If app is specified and the same as the workspace app, don't throw an error.
+	// If app is not the same as the workspace app, either
+	//   a) there is no WS app and it's specified erroneously, in which case we should error
+	//   b) there is a WS app and the flag has been set to a different app, in which case we should error.
+	// The app flag defaults to the WS app so there's an edge case where it's possible to specify
+	// `copilot task delete --app ws-app --default` and not error out, but this should be taken as
+	// specifying "default".
+	if o.app != tryReadingAppName() {
+		return fmt.Errorf("cannot specify both `--app` and `--default`")
+	}
+
+	if o.env != "" {
+		return fmt.Errorf("cannot specify both `--env` and `--default`")
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) validateEnvName() error {
+	if _, err := o.targetEnv(); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (o *deleteTaskOpts) validateAppName() error {
+	if _, err := o.store.GetApplication(o.app); err != nil {
+		return fmt.Errorf("get application: %w", err)
+	}
+	return nil
+}
+
+func (o *deleteTaskOpts) targetEnv() (*config.Environment, error) {
+	env, err := o.store.GetEnvironment(o.app, o.env)
+	if err != nil {
+		return nil, fmt.Errorf("get environment %s config: %w", o.env, err)
+	}
+	return env, nil
+}
```

Comment:
Hmm can we ditch those wrapping methods for readability? I feel

```go
func (o *deleteTaskOpts) validateFlagsWithEnv() error {
	if o.app != "" {
		// get app from store
	}

	if o.env != "" {
		// get env from store
	}

	return nil
}
```
is what we usually do.